sycl/test/usm/allocatorll.cpp

```diff
@@ -35,23 +33,21 @@ int main() {
   Node *d_cur = nullptr;
   Node h_cur;
 
-  for (int i = 0; i < numNodes; i++) {
-    if (i == 0) {
-      d_head = alloc.allocate(1);
-      if (d_head == nullptr) {
-        failed = true;
-        break;
-      }
-      d_cur = d_head;
-    }
+  try {
+    d_head = alloc.allocate(1);
+  } catch (memory_allocation_error e) {
+    return -1;
+  };
+
+  d_cur = d_head;
 
+  for (int i = 0; i < numNodes; i++) {
     h_cur.Num = i * 2;
 
     if (i != (numNodes - 1)) {
       h_cur.pNext = alloc.allocate(1);
       if (h_cur.pNext == nullptr) {
```

Comment:
Please, remove this `if` statement. 
`allocate` will throw.

Reply:
Fixed.